From 3328547c840664896c12d0fd2db8f63e3a398a62 Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Wed, 16 Mar 2016 18:59:23 -0700
Subject: [PATCH 1/3] xextproto 7.1 support update to the source code

While the compilation script was updated, for some reason, the source
code did not get the update.

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
Reviewed-by: Connor Behan <connor.behan@gmail.com>
---
 src/r128_crtc.c   | 6 ++++++
 src/r128_output.c | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/src/r128_crtc.c b/src/r128_crtc.c
index d62cc6a..ed77492 100644
--- a/src/r128_crtc.c
+++ b/src/r128_crtc.c
@@ -35,7 +35,13 @@
 
 #include "xf86.h"
 #include "xf86Modes.h"
+
+#ifdef HAVE_XEXTPROTO_71
 #include "X11/extensions/dpmsconst.h"
+#else
+#define DPMS_SERVER
+#include "X11/extensions/dpms.h"
+#endif
 
 #include "r128.h"
 #include "r128_probe.h"
diff --git a/src/r128_output.c b/src/r128_output.c
index a2c754b..73b4af2 100644
--- a/src/r128_output.c
+++ b/src/r128_output.c
@@ -35,7 +35,13 @@
 
 #include "xf86.h"
 #include "xf86Modes.h"
+
+#ifdef HAVE_XEXTPROTO_71
 #include "X11/extensions/dpmsconst.h"
+#else
+#define DPMS_SERVER
+#include "X11/extensions/dpms.h"
+#endif
 
 #include "r128.h"
 #include "r128_probe.h"
-- 
2.10.2


From a589dd6baf7fa26f8ea7008ba6bd533a5ba3e76e Mon Sep 17 00:00:00 2001
From: Connor Behan <connor.behan@gmail.com>
Date: Wed, 13 Apr 2016 12:41:00 -0400
Subject: [PATCH 2/3] Remove gatos advertisement

There are many current working ways to use a TV tuner with Linux. This
is not one of them.
Fixes: https://bugs.freedesktop.org/show_bug.cgi?id=94915

Signed-off-by: Connor Behan <connor.behan@gmail.com>
---
 src/r128_driver.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/r128_driver.c b/src/r128_driver.c
index 9a676d5..5695614 100644
--- a/src/r128_driver.c
+++ b/src/r128_driver.c
@@ -1435,10 +1435,6 @@ Bool R128PreInit(ScrnInfoPtr pScrn, int flags)
     if (info->MMIO) R128UnmapMMIO(pScrn);
     info->MMIO = NULL;
 
-    xf86DrvMsg(pScrn->scrnIndex, X_NOTICE,
-	"For information on using the multimedia capabilities\n\tof this"
-	" adapter, please see http://gatos.sf.net.\n");
-
     return TRUE;
 
   fail:
-- 
2.10.2


From 5ab23b3a8ef18ab9ff96ac8be174380d36e185e5 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 19 Jul 2016 10:03:56 -0400
Subject: [PATCH 3/3] Adapt Block/WakeupHandler signature for ABI 23

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/compat-api.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/compat-api.h b/src/compat-api.h
index 6bc946f..89976e4 100644
--- a/src/compat-api.h
+++ b/src/compat-api.h
@@ -75,8 +75,13 @@
 
 #define SCREEN_INIT_ARGS_DECL ScreenPtr pScreen, int argc, char **argv
 
+#if ABI_VIDEODRV_VERSION >= SET_ABI_VERSION(23, 0)
+#define BLOCKHANDLER_ARGS_DECL ScreenPtr arg, pointer pTimeout
+#define BLOCKHANDLER_ARGS arg, pTimeout
+#else
 #define BLOCKHANDLER_ARGS_DECL ScreenPtr arg, pointer pTimeout, pointer pReadmask
 #define BLOCKHANDLER_ARGS arg, pTimeout, pReadmask
+#endif
 
 #define CLOSE_SCREEN_ARGS_DECL ScreenPtr pScreen
 #define CLOSE_SCREEN_ARGS pScreen
-- 
2.10.2

